bin/generate-gitlab-ci-yml.sh

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env bash
+
+original_directory=$(pwd)
+. "${original_directory}/share/spack/setup-env.sh"
+
+if [ -z "${DOWNSTREAM_CI_REPO}" ] ; then
+    echo "ERROR: missing variable: DOWNSTREAM_CI_REPO" >&2
+    exit 1
+fi
+
+if [ -z "${SPACK_RELEASE_ENVIRONMENT_PATH}" ] ; then
+    echo "ERROR: missing variable: SPACK_RELEASE_ENVIRONMENT_PATH" >&2
+    exit 1
+fi
+
+if [ -z "${CDASH_AUTH_TOKEN}" ] ; then
+    echo "ERROR: missing variable: CDASH_AUTH_TOKEN" >&2
+    exit 1
+fi
+
+if [ -z "${SPACK_RELEASE_ENVIRONMENT_REPO}" ] ; then
+    echo "Assuming spack repo contains environment" >&2
+    env_repo_dir=${original_directory}
+else
+    echo "Cloning environment repo: ${SPACK_RELEASE_ENVIRONMENT_REPO}" >&2
+    mkdir /work
+    cd /work
+    git clone ${SPACK_RELEASE_ENVIRONMENT_REPO} envrepo
+    cd envrepo
+    env_repo_dir=$(pwd)
+fi
+
+current_branch="$CI_COMMIT_REF_NAME"
+
+# Because want to see generated gitlab-ci file as an artifact,
+# we need to write it within the spack repo cloned by gitlab-ci.
+gen_ci_dir="${original_directory}/ci-generation"
+gen_ci_file="${gen_ci_dir}/.gitlab-ci.yml"
+mkdir -p ${gen_ci_dir}
+
+env_dir="${env_repo_dir}/${SPACK_RELEASE_ENVIRONMENT_PATH}"
+
+if [ ! -f "${env_dir}/spack.yaml" ] ; then
+    echo "ERROR: Cannot find spack environment file in ${env_dir}"
+    exit 1
+fi
+
+cd $env_dir
+
+token_file=/tmp/cdash_auth_token
```

Comment:
Put this in your generated temp dir to avoid name collisions.

Reply:
Yes, good suggestion, will do.